src/main/java/de/tum/in/www1/artemis/domain/plagiarism/PlagiarismVerdict.java

```diff
@@ -2,5 +2,26 @@
 
 public enum PlagiarismVerdict {
 
-    PLAGIARISM, POINT_DEDUCTION, WARNING, NO_PLAGIARISM
+    // Warning: The ordering of the enum values are important, they should be ordered from the most severe to least severe.
+    // See findMostSevereVerdict method below.
+    PLAGIARISM, POINT_DEDUCTION, WARNING, NO_PLAGIARISM;
+
+    /**
+     * Finds the most severe plagiarism verdict where severity is defined by the order of the enum constants in {@link PlagiarismVerdict}
+     * (The first enum constant is the most severe).
+     *
+     * In the intended usage scenario, all members of the plagiarismVerdicts should belong to the same student and in the same course or exam.
+     * @param plagiarismVerdicts an iterable of plagiarism verdicts.
+     * @return the most servere verdict for the student or null if there is none
+     */
+    public static PlagiarismVerdict findMostSevereVerdict(Iterable<PlagiarismVerdict> plagiarismVerdicts) {
+        for (PlagiarismVerdict maxVerdict : PlagiarismVerdict.values()) {
+            for (PlagiarismVerdict currentVerdict : plagiarismVerdicts) {
+                if (currentVerdict == maxVerdict) {
+                    return currentVerdict;
+                }
+            }
+        }
+        return null;
+    }
```

Comment:
If I see this correctly, this should work as well. Note that this changes the signature of the method, but right now it is only used with Lists so this should be fine.
```suggestion
    public static PlagiarismVerdict findMostSevereVerdict(List<PlagiarismVerdict> plagiarismVerdicts) {
        return plagiarismVerdicts.stream().max(Comparator.comparing(PlagiarismVerdict::ordinal)).orElse(null);
    }
```

Reply:
You are right that we currently only use this with a list but I changed the parameter type to `Collection<PlagiarismVerdict>` to still not lose generality more than necessary. Also I needed to add `.filter(Objects::nonNull)`. I didn't know about `Enum::ordinal`, that was nice to learn, thanks!